### PR TITLE
Public feed

### PR DESCRIPTION
**Pull request by jajanet**
got basic implementation of fetching msgs going

### CODE REVIEW

File: .factorypath [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<factorypath>
```
Comment (tylerhou) [inline .factorypath]: This file is an editor-specific configuration file and is not needed by other developers, so please add this file to the .gitignore: https://help.github.com/en/articles/ignoring-files

You can do so by adding the following lines to .gitignore:
```
# Java annotation processor (APT)
.factorypath
```

After changing .gitignore, to remove the file, run `git rm --cached .factorypath`

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -46,19 +48,42 @@ public void storeMessage(Message message) {
     datastore.put(messageEntity);
   }
 
+  /** Get all the messages currently in the Datastore */
```
Comment (tylerhou) [inline src/main/java/com/google/codeu/data/Datastore.java]: Nit: period (`Datastore. */`)

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import com.google.codeu.data.Message;
+import com.google.gson.Gson;
+
+
+/**
+ * Handles fetching all messages for the public feed.
+ */
+
+@WebServlet("/feed")
+public class MessageFeedServlet extends HttpServlet{
```
Comment (tylerhou) [inline src/main/java/com/google/codeu/servlets/MessageFeedServlet.java]: Please keep style consistent with the other files:

`HttpServlet {`
Two spaces for indentation.

File: src/main/webapp/feed.html [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  <title>Message Feed</title>
+  <link rel="stylesheet" href="/css/main.css">
+  <link rel="stylesheet" href="/css/user-page.css">
+
+  <script>
+    // Fetch messages and add them to the page.
+    function fetchMessages() {
+      const url = '/feed';
+      fetch(url).then((response) => { return response.json() })
+      .then((messages) => {
```
Comment (tylerhou) [inline src/main/webapp/feed.html]: Lines which wrap over multiple lines (including chained methods) should be indented:

```
fetch(url)
  .then(...)
  .then(...)
```

File: src/main/webapp/feed.html [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        const messageContainer = document.getElementById('message-container');
+        messageContainer.innerHTML = (messages.length == 0) 
+        ? '<p>There are no posts yet.</p>' : messageContainer.innerHTML = '';
```
Comment (tylerhou) [inline src/main/webapp/feed.html]: Same here:

```
messageContainer.innerHTML = (messages.length == 0)
  ? '<p>There are no posts yet.</p>' : messageContainer.innerHTML = '';
```

File: src/main/webapp/feed.html [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+        messages.forEach((message) => {
+          const messageDiv = buildMessageDiv(message);
+          messageContainer.appendChild(messageDiv);
+        });
+      })
+      .catch((error) => {
+        console.log(error.message)
+        });
```
Comment (tylerhou) [inline src/main/webapp/feed.html]: Closing brackets should be on the same column as the beginning of the line with the opening bracket:

```
.catch((error) => {
  ...
});
```

File: src/main/webapp/feed.html [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+      const messageDiv = document.createElement('div');
+      messageDiv.classList.add("message-div");
+      messageDiv.appendChild(headerDiv);
+      messageDiv.appendChild(bodyDiv);
+
+      return messageDiv;
+    }
+
+    // Fetch data and populate the UI of the page.
+    function buildUI() {  fetchMessages();  }
```
Comment (tylerhou) [inline src/main/webapp/feed.html]: Please remove unnecessary line breaks in this file:

```
    // Fetch data and populate the UI of the page.
    function buildUI() {  fetchMessages();  }
   </script>
 </head>
 <body onload="buildUI()">
   <div id="content">
     <h1>Message Feed</h1>
     ...
```

File: src/main/webapp/feed.html (hunk 4 above)
Comment (tylerhou) [reply src/main/webapp/feed.html]: (You can also install a code formatter like Prettier to fix this automatically.)

File: src/main/java/com/google/codeu/servlets/UserListServlet.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+ * Handles fetching all users for the community page.
+ */
+@WebServlet("/user-list")
+public class UserListServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+
+    response.getOutputStream().println("this will be my user list");
```
Comment (tylerhou) [inline src/main/java/com/google/codeu/servlets/UserListServlet.java]: I would also suggest following the same strategy I wrote in Thomas' PR: https://github.com/jajanet/team-08-codeu/pull/5/files#r288369916

File: .factorypath (hunk 1 above)
Comment (jajanet) [reply .factorypath]: thanks! i wasn't sure what this file was honestly but good to know for the future

File: src/main/java/com/google/codeu/servlets/UserListServlet.java (hunk 8 above)
Comment (jajanet) [reply src/main/java/com/google/codeu/servlets/UserListServlet.java]: thanks

**Pull request merged**